### PR TITLE
SQL DQL fluent API

### LINKED ISSUE #81: Implement fluent API for DQL SQL statements
## Overview
Implement the SQL class in the `org.fungover.breeze.sql` package to allow the user to create DQL SQL statements through a fluent API

### Class Structure
* Create `SQL` class
* Create subclasses and interfaces to return the correct functions

## Documentation Requirements
- Complete Javadoc for all public methods and classes

## Test Requirements
- Unit tests for all public methods
- Minimum 90% code coverage

## Implementation Guidelines
- Use final classes/fields where appropriate
- Follow Java naming conventions
- Keep methods focused and small

## Dependencies
- No external runtime dependencies allowed
- JUnit and AssertJ permitted for testing

## Definition of Done
- [X] All required methods implemented
- [x] Complete Javadoc documentation
- [X] All tests passing with required coverage
- [X] No compiler warnings
- [X] Passes static code analysis

### PR DESCRIPTION
**Pull request by voffie**
Merging this PR adds the fluent SQL DQL API functionality. Key features and improvements implemented in this PR include:

* **Core Functionality:**
  * Added SQL class
  * Added subclasses & interfaces with specified methods
* **Testing:**
  * Implemented unit & integration tests to ensure code reliability.

## Issues Closed
This PR closes the following issues:

Closes #81: Implement fluent API for DQL SQL statements

## Summary by CodeRabbit

- **New Features**
	- Introduced a fluent interface for building customizable SQL queries. Users can now chain methods to construct queries with selections, filtering, joins, grouping, ordering, and limits.

- **Tests**
	- Added a comprehensive suite of tests to validate the query builder's functionality across various scenarios and edge cases, ensuring reliable and robust operations.

### CODE REVIEW

File: src/main/java/org/fungover/breeze/sql/SQL.java [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+     */
+    ConditionStep or(String column);
+
+    /**
+     * Adds the query to select on
+     *
+     * @param query The query to equal a column by
+     * @return The supported functions following DQL SQL
+     */
+    EqualStep equalTo(String query);
```
Comment (kappsegla) [inline src/main/java/org/fungover/breeze/sql/SQL.java:199]: Would using the name isEqualTo read better in the fluent api?

File: src/main/java/org/fungover/breeze/sql/SQL.java [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+    String build();
+
+    /**
+     * Adds support for ordering the result based on a column and which order
+     *
+     * @param column    The column to order by
+     * @param ascending If the order should be ascending or not
+     * @return The supported functions following DQL SQL
+     */
+    OrderByStep orderBy(String column, boolean ascending);
```
Comment (kappsegla) [inline src/main/java/org/fungover/breeze/sql/SQL.java:325]: Instead of using a boolean here couldn't we have an Enum value with DESC and ASC or the longer versions Descending / Ascending? Makes it easier to read. Could maybe use something like orderBy("column").Desc()  where we could skip the appended method for sorting order and by default get the ascending order if not specifying it.

**Review by kappsegla — COMMENTED** [review]
Added two smaller suggestions about naming things. This is probably an api that would need a couple of iterations in the future to find the best form and names for easy an intuitive use.

File: src/main/java/org/fungover/breeze/sql/SQL.java [hunk 3]
```diff
@@ -0,0 +1,628 @@
...
+     * @return The entire SQL statement
+     */
+    String build();
+
+    /**
+     * Adds support for condition steps to the SQL statement
+     *
+     * @return The supported functions following DQL SQL
+     */
+    WhereStep where();
```
Comment (VictoriaNaenfeldtEklund) [inline src/main/java/org/fungover/breeze/sql/SQL.java:90]: Very good job on this! I just have a small question / suggestion, a queggestion :) ! 
Looking at the methods, they often take a String column/int limit/String condition as a parameter. 
Could the where() method maybe do this as well to be more "inline" with the syntax of the other methods?

File: src/main/java/org/fungover/breeze/sql/SQL.java (hunk 2 above)
Comment (voffie) [reply src/main/java/org/fungover/breeze/sql/SQL.java:325]: That could work! Will take a look at it later this week!

File: src/main/java/org/fungover/breeze/sql/SQL.java (hunk 1 above)
Comment (voffie) [reply src/main/java/org/fungover/breeze/sql/SQL.java:199]: That is true! As you said above it probably requires a couple of iterations, and possibly also some beta testers to find the "perfect" naming convention

File: src/main/java/org/fungover/breeze/sql/SQL.java (hunk 3 above)
Comment (voffie) [reply src/main/java/org/fungover/breeze/sql/SQL.java:90]: It probably could! It would remove the need for the other functions in this case, which might not be the best move. I am up for discussing this more in person later this week!

**Review by EvgeniiaGranina — APPROVED** [review]

**Review by EvgeniiaGranina — APPROVED** [review]

**Comment by SeydaKinaci** [comment]
Your code is very clear and easy to read. The Javadocs are also well-written, great job!

**Review by SeydaKinaci — APPROVED** [review]
Very clear and understandable code. Great job!
